public/app/features/explore/ExplorePaneContainer.tsx

```diff
@@ -0,0 +1,124 @@
+import React from 'react';
+import { hot } from 'react-hot-loader';
+import { compose } from 'redux';
+import { connect, ConnectedProps } from 'react-redux';
+import memoizeOne from 'memoize-one';
+import { withTheme } from '@grafana/ui';
+import { DataQuery, ExploreUrlState, EventBusExtended, EventBusSrv } from '@grafana/data';
+import { selectors } from '@grafana/e2e-selectors';
+import store from 'app/core/store';
+import { lastSavedUrl } from './state/main';
+import { initializeExplore, refreshExplore } from './state/explorePane';
+import { ExploreId } from 'app/types/explore';
+import { StoreState } from 'app/types';
+import {
+  DEFAULT_RANGE,
+  ensureQueries,
+  getTimeRange,
+  getTimeRangeFromUrl,
+  lastUsedDatasourceKeyForOrgId,
+  parseUrlState,
+} from 'app/core/utils/explore';
+import { getTimeZone } from '../profile/state/selectors';
+import Explore from './Explore';
+
+type PropsFromRedux = ConnectedProps<typeof connector>;
+interface Props extends PropsFromRedux {
+  exploreId: ExploreId;
+  split: boolean;
+}
+
+export class ExplorePaneContainerUnconnected extends React.PureComponent<Props & ConnectedProps<typeof connector>> {
+  el: any;
+  exploreEvents: EventBusExtended;
+
+  constructor(props: Props) {
+    super(props);
+    this.exploreEvents = new EventBusSrv();
+    this.state = {
+      openDrawer: undefined,
+    };
+  }
+
+  componentDidMount() {
+    const { initialized, exploreId, initialDatasource, initialQueries, initialRange, originPanelId } = this.props;
+    const width = this.el ? this.el.offsetWidth : 0;
+
+    // initialize the whole explore first time we mount and if browser history contains a change in datasource
+    if (!initialized) {
+      this.props.initializeExplore(
+        exploreId,
+        initialDatasource,
+        initialQueries,
+        initialRange,
+        width,
+        this.exploreEvents,
+        originPanelId
+      );
+    }
+  }
+
+  componentWillUnmount() {
+    this.exploreEvents.removeAllListeners();
+  }
+
+  componentDidUpdate(prevProps: Props) {
+    this.refreshExplore(prevProps.urlQuery);
+  }
+
+  refreshExplore = (prevUrlQuery: string) => {
+    const { exploreId, urlQuery } = this.props;
+
+    // Update state from url only if it changed and only if the change wasn't initialised by redux to prevent any loops
+    if (urlQuery !== prevUrlQuery && urlQuery !== lastSavedUrl[exploreId]) {
+      this.props.refreshExplore(exploreId, urlQuery);
+    }
+  };
+
+  getRef = (el: any) => {
+    this.el = el;
+  };
+
+  render() {
+    const exploreClass = this.props.split ? 'explore explore-split' : 'explore';
+    return (
+      <div className={exploreClass} ref={this.getRef} aria-label={selectors.pages.Explore.General.container}>
+        {this.props.initialized && <Explore exploreId={this.props.exploreId} />}
```

Comment:
We wait until state is initialized. Should not take long, only async thing is loading the data source so I don't think this needs loading state but could be added later.